binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,45 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        if value < self.value:
+            if not self.left:
+                self.left = BinarySearchTree(value)
+            else:
+                self.left.insert(value)
+        elif value >= self.value:
+            if not self.right:
+                self.right = BinarySearchTree(value)
+            else:
+                self.right.insert(value)
+        pass
 
-  def contains(self, target):
-    pass
+    def contains(self, target):
+        if target < self.value:
+            if self.left is None:
+                return False
+            else:
+                return self.left.contains(target)
```

Comment:
the syntax is weird right? it seems like what you are saying is the left leaf on that node has the target, but you are actually recursively calling this again until the base case is fulfilled ( you got to the left most node in the tree). Excellent work, and this shows the difficulty of naming variables/functions.